client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -442,19 +496,25 @@ class DeltaSharingRestClient(
   // Send paginated queryTable requests. Loop internally to fetch and concatenate all pages,
   // then return (version, respondedFormat, actions, refreshToken) tuple.
   private def getFilesByPage(
+      table: Table,
       targetUrl: String,
       request: QueryTableRequest): (Long, String, Seq[String], Option[String]) = {
     val allLines = ArrayBuffer[String]()
     val start = System.currentTimeMillis()
     var numPages = 1
 
     // Fetch first page
-    var updatedRequest = request.copy(maxFiles = Some(maxFilesPerReq))
-    val (version, respondedFormat, lines) = getNDJson(targetUrl, updatedRequest)
+    val updatedRequest = request.copy(
+      maxFiles = Some(maxFilesPerReq),
+      pageToken = request.pageToken)
+
+    val (version, respondedFormat, lines, queryIdOpt)
+    = getNDJsonWithAsync(table, targetUrl, updatedRequest)
```

Comment:
Same question. Should we do an if (enableAsync) here as well?

Reply:
+1